homeassistant/components/fronius/sensor.py

```diff
@@ -136,62 +150,139 @@ def device_state_attributes(self):
             _LOGGER.error("Failed to update: invalid response returned."
                           "Maybe the configured device is not supported")
 
-        if values:
-            self._state = values['status']['Code']
-            attributes = {}
-            for key in values:
-                if 'value' in values[key]:
-                    attributes[key] = values[key].get('value', 0)
-            self._attributes = attributes
+        if not values:
+            return
+        attributes = self._attributes
+        # Copy data of current fronius device
+        for key, entry in values.items():
+            # If the data is directly a sensor
+            if 'value' in entry:
+                attributes[key] = entry
+            # Handle system overview with multiple sensors
+            elif '1' in entry or '0' in entry:
+                for index in values:
+                    attributes['{}_{}'.format(
+                        key, index
+                    )] = values[index]
+        self._attributes = attributes
+
+        # Add discovered value fields as sensors
+        # because some fields are only sent temporarily
+        new_sensors = []
+        for key in attributes:
+            if key not in self.sensors:
+                self.sensors.add(key)
+                _LOGGER.info(
+                    "Discovered %s, adding as sensor.",
+                    key
+                )
+                new_sensors.append(FroniusTemplateSensor(self, key))
+        self._add_entities(new_sensors, True)
+
+        # Schedule an update for all included sensors
+        for sensor in self._registered_sensors:
+            sensor.async_schedule_update_ha_state(True)
 
     async def _update(self):
         """Return values of interest."""
         pass
 
+    async def register(self, sensor):
+        """Register child sensor for update subscriptions."""
+        self._registered_sensors.add(sensor)
+
 
-class FroniusInverterSystem(FroniusSensor):
+class FroniusInverterSystem(FroniusAdapter):
```

Comment:
Please don't have the entities inherit the adapter. Please separate the entity from the instance that handles the device data and updates.

Reply:
I see these are not entities anymore. But it's confusing since we call them sensor classes and keep some entity specific properties like `device_state_attributes`. Maybe it's enough to rename some things about these to make it clear that they are not entities?

Reply:
I consistently renamed them adapters now, that should make things clearer.

Reply:
Great! Thanks!